### DISCUSSION TITLE
Renovate can't reopen autoclosed PRs on BitBucket.org even with recreateWhen: 'always'

### DISCUSSION THREAD

**Discussion by olegafx** [renovatebot/renovate#32889] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

BitBucket.org, v39.4.0

### Please tell us more about your question or problem

There was a moment when `Renovate` autoclosed all the open PRs on `BitBucket.org`, and it hasn't been able to create new ones since. I checked the log and saw that it fails right after finding that a PR already exists:
```
DEBUG: Found existing branch PR
DEBUG: PR has been closed or merged since this run started - aborting
DEBUG: Passing repository-changed error up
INFO: Repository has changed during renovation - aborting
```
I examined the source code and found [where](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/branch/index.ts#L272) this is happening. The weird thing is that there is no check for `recreateWhen`. I thought it should be recreated even when closed with `recreateWhen: 'always'`.
Another problem is that `BitBucket.org` doesn't allow the removal of PRs. Therefore, if `Renovate` is functioning correctly, the only option I see right now is to rename the branches.

**Comment by rarkins** [comment]
Needs more logs

**Reply by olegafx** [reply]
Added

**Reply by olegafx** [reply]
Setting `repositoryCache` to `reset` fixed the issue
